dockstore-client/src/main/java/io/dockstore/client/cli/nested/WesLauncher.java

```diff
@@ -117,6 +134,54 @@ public static Workflow getWorkflowForEntry(WorkflowClient workflowClient, String
         return workflowClient.getWorkflowsApi().getPublishedWorkflowByPath(path, WorkflowSubClass.BIOWORKFLOW.toString(), null, version);
     }
 
+    /**
+     * Provisions a workflow's associated files in a temporary local directory
+     *
+     * @param workflowClient The WorkflowClient
+     * @param workflowEntry The workflow entry (i.e. github.com/myRepo/myWorkflow:version)
+     * @param descriptorType The type of descriptor for this workflow (WDL, CWL, etc...)
+     * @return A zipped File object
+     */
+    public static File provisionFilesLocally(WorkflowClient workflowClient, String workflowEntry, String descriptorType) {
+
+        // A temporary directory which will house all downloaded content
+        File tempDir;
+
+        try {
+            tempDir = Files.createTempDirectory(DOCKSTORE_ROOT_TEMP_DIR_PREFIX).toFile().getAbsoluteFile();
+        } catch (IOException ex) {
+            exceptionMessage(ex, "Could not create a temporary working directory.", IO_ERROR);
+            throw new RuntimeException(ex);
+        }
+
+        // A zip file containing the content of the provided workflow
+        File zippedWorkflow;
+
+        // Download the contents locally
+        try {
+            zippedWorkflow = workflowClient.downloadTargetEntry(workflowEntry,
+                ToolDescriptor.TypeEnum.fromValue(descriptorType), false, tempDir);
+        } catch (IOException ex) {
+            exceptionMessage(ex, "A problem was encountered while downloading the entry.", IO_ERROR);
+            throw new RuntimeException(ex);
+        }
+
+        // The directory where the unzipped workflow contents are located
+        File unzippedWorkflowDir;
+
+        // Unzip the workflow to a nested temporary directory
+        try {
+            unzippedWorkflowDir = Files.createTempDirectory(Path.of(tempDir.getAbsolutePath()), DOCKSTORE_NESTED_TEMP_DIR_PREFIX).toFile().getAbsoluteFile();
```

Comment:
Add a line `unzippedWorkflowDir.deleteOnExit()` to clean up.

Reply:
Thanks, I thought temporary directories would cleanup on exit.

Added: https://github.com/dockstore/dockstore-cli/pull/113/commits/73b5a0bbeb3d04bf035242170002d24af323df96